src/CSET/operators/plot.py

```diff
@@ -1065,6 +1074,32 @@ def _custom_colourmap_precipitation(cube: iris.cube.Cube, cmap, levels, norm):
     return cmap, levels, norm
 
 
+def _get_num_models(cube: iris.cube.Cube | iris.cube.CubeList) -> int:
+    """Return number of models based on cube attributes."""
+    model_names = list(
+        filter(
+            lambda x: x is not None,
+            set([cb.attributes.get("model_name", None) for cb in iter_maybe(cube)]),
```

Comment:
```suggestion
            {cb.attributes.get("model_name", None) for cb in iter_maybe(cube)},
```
Directly use set comprehension to avoid building an intermediate list.

Reply:
:heavy_check_mark: